### ISSUE TITLE
ShyLU: Tests failures with Tpetra's deprecated code is disabled

### ISSUE THREAD

**Issue by tjfulle** [trilinos/Trilinos#5617]
## Bug Report
@trilinos/shylu 

Part of: #5602 

### Description

With Tpetra's deprecated code disabled, 2 ShyLU tests fail for `GlobalOrdinal=int`:

- ShyLU_DDFROSch_test_thyra_xpetra_laplace_TLP_IPOUHarmonic_DIM2_DPN2_ORD0_TPETRA_MPI_4
- ShyLU_DDFROSch_test_thyra_xpetra_laplace_TLP_IPOUHarmonic_DIM2_DPN2_ORD1_TPETRA_MPI_4

See the full configuration for `GlobalOrdinal=int` at https://github.com/trilinos/Trilinos/issues/5602#issuecomment-516593697

**Comment by tjfulle** [comment]
@kddevin can you assign the Tpetra team member that was going to look at this?

**Comment by brian-kelley** [comment]
@kddevin @tjfulle @trilinos/xpetra I have a fix for these, it was just about changing an Xpetra::MatrixFactory::Build() call with StaticProfile and the exact # entries per row. However, I noticed a couple of things that should probably be changed:

-Xpetra::CrsMatrix does not support getNumEntriesInGlobalRow(), which would be really useful for changes like this one. I had to do getGlobalRowView again for every row to read off the row sizes. It's easy to implement getNumEntriesInGlobalRow() by calling it on the underlying matrix, but I'm wondering if there's a good reason this hasn't already been done.

-Xpetra::MatrixFactory::Build still uses Xpetra::DynamicProfile as the default "profile" parameter value, even if Tpetra the underlying library.
